selenium/che-selenium-test/src/main/java/org/eclipse/che/selenium/pageobject/theia/TheiaTerminal.java

```diff
@@ -0,0 +1,166 @@
+
+  @Inject
+  private TheiaTerminal(
+      SeleniumWebDriverHelper seleniumWebDriverHelper,
+      SeleniumWebDriver seleniumWebDriver,
+      TheiaProjectTree theiaProjectTree,
+      TheiaEditor theiaEditor,
+      TheiaIde theiaIde,
+      TheiaNewFileDialog theiaNewFileDialog,
+      TheiaDeleteFileDialog theiaDeleteFileDialog) {
+    this.seleniumWebDriverHelper = seleniumWebDriverHelper;
+    this.seleniumWebDriver = seleniumWebDriver;
+    this.theiaProjectTree = theiaProjectTree;
+    this.theiaEditor = theiaEditor;
+    this.theiaIde = theiaIde;
+    this.theiaNewFileDialog = theiaNewFileDialog;
+    this.theiaDeleteFileDialog = theiaDeleteFileDialog;
+  }
+
+  public interface Locators {
+    String TERMINAL_BODY_ID_TEMPLATE = "terminal-%s";
+    String TERMINAL_TEXT_LAYER = "//canvas[@class='xterm-text-layer']";
+    String TERMINAL_UPPEST_LAYER = "//canvas[@class='xterm-cursor-layer']";
+  }
+
+  private WebElement getTerminalTextLayer() {
+    return seleniumWebDriverHelper.waitVisibility(By.xpath(TERMINAL_TEXT_LAYER));
+  }
+
+  public void waitTerminal() {
+    seleniumWebDriverHelper.waitVisibility(By.xpath(TERMINAL_TEXT_LAYER));
+  }
+
+  public void clickOnTerminalTextArea() {
+    seleniumWebDriverHelper.waitAndClick(By.xpath(TERMINAL_UPPEST_LAYER));
+  }
+
+  public void type(String expectedText) {
+    seleniumWebDriverHelper.sendKeys(expectedText);
+  }
+
+  public void performCommand(String command) {
+    type(command);
+    seleniumWebDriverHelper.sendKeys(ENTER.toString());
+  }
+
+  private void copyTerminalTextToClipboard() {
+    Dimension textLayerSize = getTerminalTextLayer().getSize();
+    final Actions action = seleniumWebDriverHelper.getAction();
+
+    final int xBeginCoordinateShift = -(textLayerSize.getWidth() / 2);
+    final int yBeginCoordinateShift = -(textLayerSize.getHeight() / 2);
+
+    seleniumWebDriverHelper.moveCursorTo(getTerminalTextLayer());
+
+    // WaitUtils.sleepQuietly(3);
+
+    seleniumWebDriverHelper
+        .getAction()
+        .moveByOffset(xBeginCoordinateShift, yBeginCoordinateShift)
+        .perform();
+
+    // WaitUtils.sleepQuietly(3);
+
+    action.clickAndHold().perform();
+    seleniumWebDriverHelper
+        .getAction()
+        .moveByOffset(textLayerSize.getWidth(), textLayerSize.getHeight())
+        .perform();
+
+    // WaitUtils.sleepQuietly(3);
+
+    action.release().perform();
+
+    String keysCombination = Keys.chord(CONTROL, INSERT);
+    seleniumWebDriverHelper.sendKeys(keysCombination);
+
+    clickOnTerminalTextArea();
+  }
+
+  public void waitTerminalOutput(String expectedText) {
+    final int timeout = LOADER_TIMEOUT_SEC;
+
+    seleniumWebDriverHelper.waitSuccessCondition(
+        driver -> {
+          copyTerminalTextToClipboard();
+          return getTerminalOutput().contains(expectedText);
+        },
+        timeout);
+  }
+
+  public String getTerminalOutput() {
+    // this is workaround and should be resolved by issue:
```

Comment:
Let's add `TODO` at the start of comment to make it clear that it is temporary solution